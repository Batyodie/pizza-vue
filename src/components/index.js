import Button from "./Button/Button.vue";
import Card from "./Card/Card.vue";
import DropDown from "./DropDown/DropDown.vue";
import Logo from "./Logo/Logo.vue";
import ShopCart from "./ShopCart/ShopCart.vue";

export { Button, Card, DropDown, Logo, ShopCart };
