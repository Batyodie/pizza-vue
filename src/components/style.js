import BtnTag from "@/components/Button/Button.css";
import Button from "@/components/Button/Button.css";
import DropDownStyle from "@/components/DropDown/DropDown.css";
import CardStyle from "@/components/Card/Card.css";
import LogoStyle from "@/components/Logo/Logo.css";
import ShopCartStyle from "@/components/ShopCart/ShopCart.css";
import BtnStyle from "@/components/Button/Button.css";

export {
  Button,
  BtnTag,
  DropDownStyle,
  CardStyle,
  LogoStyle,
  ShopCartStyle,
  BtnStyle
};
