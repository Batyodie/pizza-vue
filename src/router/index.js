import Vue from "vue";
import VueRouter from "vue-router";
import Home from "../views/Market";
import Basket from "../views/Basket/Basket";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/basket",
    name: "Basket",
    component: Basket,
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  hash: false,
  routes,
});

export default router;
