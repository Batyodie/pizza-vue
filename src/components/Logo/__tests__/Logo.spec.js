import { createLocalVue, mount } from "@vue/test-utils";
// dropdwons componens files
import { Logo } from "@/components";

import VueRouter from "vue-router";
import router from "@/router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("Unit tests for Logo component", () => {
  // fake props
  const sloganText = "Some fake title";
  const LOGO = "/img/pizza-logo.svg";
  const wrapper = mount(Logo, {
    localVue,
    router,
    propsData: {
      sloganText
    }
  });
  wrapper.setData({ logo: LOGO });

  it("should initialize correctly", () => {
    // correctly Vue instance
    expect(wrapper.exists()).toBeTruthy();
    // We expect the wrapper to be the Logo component
    expect(wrapper.is(Logo)).toBeTruthy();
    //  We expect the wrapper props to equal the fake props
    expect(wrapper.vm.sloganText).toEqual(sloganText);
    // add logo to data() component
    expect(wrapper.vm.logo).toEqual(LOGO);
    expect(wrapper.find(".Image").attributes("src")).toBe(LOGO);
  });

  it("should render correctly logo title", () => {
    const renderSlogan = `${sloganText}`;

    // Expect the wrapper to be the slogan text render
    expect(wrapper.text(renderSlogan)).toBeTruthy();
    // Expect thatrender SloganText in Slogan selector equal to renderSlogan
    expect(wrapper.find(".Slogan > p").text()).toEqual(renderSlogan);
  });
  it("should render correctly logo title to be false", () => {
    const sloganText = "";
    const wrapper = mount(Logo, {
      localVue,
      router,
      propsData: {
        sloganText
      }
    });
    const renderSloganShallow = "самая вкусная пицца во вселенной";
    // Expect the wrapper to be the slogan shallow  text render
    expect(wrapper.text(renderSloganShallow)).toBeTruthy();
    // Expect thatrender SloganText in Slogan selector equal to shallow slogan
    expect(wrapper.find(".Slogan > p").text()).toEqual(renderSloganShallow);
  });

  it("Should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
