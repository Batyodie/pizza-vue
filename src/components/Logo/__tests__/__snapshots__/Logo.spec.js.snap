// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Logo component Should match the snapshot 1`] = `
<a href="/" aria-current="page" class="router-link-exact-active router-link-active" style="text-decoration: none;">
  <div class="Inner"><img src="/img/pizza-logo.svg" width="38" height="38" alt="Main logo" class="Image">
    <div class="Slogan">
      <h1 class="SloganTitle">VUE PIZZA</h1>
      <p class="SloganText">Some fake title</p>
    </div>
  </div>
</a>
`;
